docs/combinatorics.rst

```diff
@@ -1,6 +1,116 @@
+.. currentmodule:: tskit
 .. _sec_combinatorics:
 
-=====================================
-Ranking and Unranking Tree Topologies
-=====================================
-TODO
+=============
+Combinatorics
+=============
+tskit uses a combinatorial approach to identify unique topologies of
+rooted, leaf-labelled trees. It provides methods for enumerating
+all possible tree topologies, as well as converting back and forth
+between a tree and its position, or rank, in the enumeration of all
+possible topologies. These methods do not only apply to binary trees;
+rather, they cover general, rooted trees with no unary nodes.
+
+=================================     =====================================
+:meth:`Tree.rank`                     Return the rank of this tree.
+:meth:`Tree.unrank`                   Return a Tree given its rank and
+                                      a number of leaves.
+:func:`tskit.all_tree_topologies`     Return a generator over all
+                                      leaf-labelled trees of n leaves.
+:func:`tskit.all_tree_shapes`         Return a generator over all
+                                      tree shapes of n leaves.
+:func:`tskit.all_tree_labellings`     Return a generator over all
+                                      labellings of the given tree's shape.
+=================================     =====================================
```

Comment:
@jeromekelleher Went with this naming pattern but happy to change it up if you have a better suggestion.

Reply:
I was thinking of making "topologies" synonymous with "tree shapes" , but I think you're probably right. How about ``all_tree_topologies`` -> ``all_trees``, and then keep the other two as they are? Usually when someone thinks of a tree in this context it'll be labelled.